### PR TITLE
core: add useNonceForAddress hook

### LINKED ISSUE #470: Add getNonceForAddress hook
I want get Nonce For Address

### PR DESCRIPTION
**Pull request by jaipaljadeja**
- Add a new `useNonceForAddress` hook which fetches the nonce associated with the given address in the given block
- Resolves #470

### CODE REVIEW

File: biome.json [hunk 1]
```diff
@@ -1,8 +1,5 @@
 {
   "$schema": "./node_modules/@biomejs/biome/configuration_schema.json",
-  "organizeImports": {
-    "enabled": false
```
Comment (fracek) [inline biome.json:4]: Why this change?

Comment (jaipaljadeja) [reply biome.json:4]: it was a duplicate, we already have this property below in the same file.

**Review by fracek — COMMENTED** [review]
Looks good. Just a small question and then I'll merge.

File: biome.json (hunk 1 above)
Comment (fracek) [reply biome.json:4]: Ahh thanks!
